src/compiler/debug.ts

```diff
@@ -638,7 +638,7 @@ export namespace Debug {
                         this.flags & TypeFlags.BigIntLiteral ? `LiteralType ${(this as BigIntLiteralType).value.negative ? "-" : ""}${(this as BigIntLiteralType).value.base10Value}n` :
                         this.flags & TypeFlags.UniqueESSymbol ? "UniqueESSymbolType" :
                         this.flags & TypeFlags.Enum ? "EnumType" :
-                        this.flags & TypeFlags.Intrinsic ? `IntrinsicType ${(this as IntrinsicType).intrinsicName}` :
+                        this.flags & TypeFlags.Intrinsic ? `IntrinsicType ${(this as IntrinsicType).intrinsicName}${(this as IntrinsicType).debugIntrinsicName ? ` (${(this as IntrinsicType).debugIntrinsicName})` : ""}` :

```

Comment:
How hard it would be to make `__debugTypeToString()` print this too?

Reply:
Hm, the problem is that `__debugTypeToString` goes through `typeToString`, where you kinda want to see what the user would see. But it could potentially be done.

Reply:
Maybe `TypeFormatFlags.Debug` could be added so it could be passed down to `typeToString` by `__debugTypeToString`? Those flags are not all exactly limited to formatting-concerns right now ;p

Reply:
That was my first thought but then I looked at it and got scared trying to decide if I had to mirror the flag in `NodeBuilderFlags` and also if there was even room in the limited number of bits we have + if I'm allowed to overlap... But I think it's doable.

Reply:
Yeah, it has to be in `NodeBuilderFlags`, which is full effectively full, and then to be a part of `NodeBuilderFlagsMask` so it can be the same in both flags enums, it can't be internal. Not great.

Reply:
Perhaps not worth it then right now. What is nice about `typeToString` though is that it's recursive and those types might sometimes appear at some nested positions.